### ISSUE TITLE
messages not published at all

### ISSUE THREAD

**Issue by gitunit** [uml-robotics/ros_tango_native_stream#5]
i've compiled the yellowstone branch successfully and the app starts.
however messages never get published, rqt_topic shows the topics but bandwidth and Hz says "unknown".
also i realized that the callback in the ImagePublisher is never called.
though i saw once some debugging stuff in the logcat but that happened only once and still there were no messages published. currently i would only need raw images, preferably fisheye.
do you know what could be the issue here?

**Comment by carlosjoan91** [comment]
Color images are a work in progress and they are not being published in the
current version. However, are you not getting any messages published at all
(I.e. depth and pose)? If so, any exceptions while that happens? Are you
placing the tango library in the specified folder?

**Comment by gitunit** [comment]
yes, no messages arrive. i placed everything as instructed. the header file i put first into the specified folder with a subfolder "include" but that didn't compile so i put both files into the folder without the "include" subfolder and then it compiled.

**Comment by carlosjoan91** [comment]
Do you get any exceptions on android studio?

2016-01-06 16:35 GMT-06:00 gitunit <notifications@github.com>:

**Comment by gitunit** [comment]
seldom but these are resolved after the usual reboot

**Comment by gitunit** [comment]
i restarted the application again and now it seems to do stuff due to logcat.
seeing debug strings with TAG "log-per-line" and "st-log"

above is the following, could be relevant:

```
01-07 00:10:53.827 188-677/? E/tango: calibration-provider.cc:180 Online calibration file provided, but does not exist: /sdcard/config/online-calibration.xml. Reverting to factory calibration.
01-07 00:10:53.828 188-677/? I/tango: tango-hal-device-wrapper.cc:61 Failed to open libtango_hal.so: dlopen failed: library "libtango_hal.so" not found
01-07 00:10:53.828 188-677/? E/tango: tango-hal-sensor-graph.cc:85 Could not open device shared library!
01-07 00:10:53.828 188-677/? I/tango: application-base.cc:104 Using sensor graph.
01-07 00:10:53.828 188-677/? I/tango: ocv-freak-feature-tracker.cc:57 Max keypoints which will be sent for descriptor extraction after pruning: 800
01-07 00:10:53.828 188-677/? I/tango: ocv-freak-feature-tracker.cc:62 Number of stages in which frame-frame matching will be run: 2
01-07 00:10:53.828 188-677/? W/tango: gpu-dog-pyramid.cc:104 CUDA has an error prior to starting module: invalid argument
01-07 00:10:53.828 188-677/? I/tango: ocv-freak-feature-tracker.cc:57 Max keypoints which will be sent for descriptor extraction after pruning: 800
01-07 00:10:53.828 188-677/? I/tango: ocv-freak-feature-tracker.cc:62 Number of stages in which frame-frame matching will be run: 2
01-07 00:10:53.829 188-677/? I/tango: native-sensors.cc:84 Have calibrated gyro
01-07 00:10:53.831 171-1013/? I/Camera2ClientBase: Camera 0: Opened
01-07 00:10:53.832 171-1013/? I/ProCamera2Client: ProCamera 0: Opened
```

**Comment by carlosjoan91** [comment]
Seems it's a problem with the tango library loading, does the version you
placed in the folder match the version your tango (And Project Tango Core
app) is at? Can you run other project Tango apps? (The last tested version
is Zeno, but I see nothing on the release notes that would imply more
recent versions would break it).

2016-01-06 17:17 GMT-06:00 gitunit <notifications@github.com>:

**Comment by gitunit** [comment]
i use the latest version capella and all tango apps work. tango core is also latest and should match.

**Comment by carlosjoan91** [comment]
I would need to test it myself with the latest version in that case, but I
don't have access to the tablet for around 2 weeks. In the meantime other
issues I can think of is general ROS troubleshooting. (e.g. times being
synchronized, having a good network connection to master, etc.).

2016-01-06 17:41 GMT-06:00 gitunit <notifications@github.com>:

**Comment by gitunit** [comment]
ok, i see. ROS in general works for me as i've other applications running flawlessly. maybe i can try using the capella support library files.

**Comment by carlosjoan91** [comment]
Just tested in the tablet, with it updated to the latest OTA, latest project tango core, and latest C API extracted in the folder, and I get both odometry and depth running. This is a newer release (Electra), released after the last time we spoke. Can you try with this one and let me know how it goes?

**Comment by gitunit** [comment]
still nothing. topics are advertised but never actually published. :disappointed:

**Comment by nuclearmistake** [comment]
Smells like DNS. Try export ROS_HOSTNAME="IP address" in the subscriber's shell before running it... With the IP of the machine running the subscriber on the right hand of the assignment.

If the tango cannot figure out the IP of the subscriber to open the TCP transport, it won't establish the connection.

**Comment by gitunit** [comment]
tried that, nothing changed except now i see some log messages about features ..but nothing inside rviz or rqt_topic. the ROS_HOSTNAME should just contain the ip without the port of the pc right?

**Comment by carlosjoan91** [comment]
Yes, ROS_HOSTNAME should be the ip without the port, also make sure you
have set your ROS_MASTER_URI to the actual machine IP, and not to
localhost.

2016-01-25 18:44 GMT-05:00 gitunit <notifications@github.com>:

**Comment by nuclearmistake** [comment]
What does rqt_graph show?

**Comment by gitunit** [comment]
@carlosjoan91 that didnt help @nuclearmistake this is what i get from rqt_graph: ![rqt_graph](http://i.imgur.com/hJ4rJZM.png)

**Comment by nuclearmistake** [comment]
Can you check the "dead topics" and "dead sinks" boxes, and get a 
screenshot of rqt_graph with the subscribers also running?

On 1/25/2016 7:15 PM, gitunit wrote:
>
> @carlosjoan91 <https://github.com/carlosjoan91> that didnt help 
> @nuclearmistake <https://github.com/nuclearmistake> this is what i get 
> from rqt_graph: rqt_graph 
> <https://camo.githubusercontent.com/acf8a8053e278055d612acc0ddb19e87f84b54ac/687474703a2f2f692e696d6775722e636f6d2f684a34724a5a4d2e706e67>
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/uml-robotics/ros_tango_native_stream/issues/5#issuecomment-174740579>.
>

-- 
--
Eric McCann
    University of Massachusetts, Lowell
    Department of Computer Science
    One University Avenue
    Olsen Hall, Room 304
    Lowell, MA  01854
Lab:             978.934.3385
Email:           emccann@cs.uml.edu
Homepage (lab):  www.cs.uml.edu/robots
Homepage:        www.emccann.net

**Comment by nuclearmistake** [comment]
And paste the output of "rosnode info [the subscriber node's name]"?

On 1/25/2016 7:15 PM, gitunit wrote:
>
> @carlosjoan91 <https://github.com/carlosjoan91> that didnt help 
> @nuclearmistake <https://github.com/nuclearmistake> this is what i get 
> from rqt_graph: rqt_graph 
> <https://camo.githubusercontent.com/acf8a8053e278055d612acc0ddb19e87f84b54ac/687474703a2f2f692e696d6775722e636f6d2f684a34724a5a4d2e706e67>
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/uml-robotics/ros_tango_native_stream/issues/5#issuecomment-174740579>.
>

-- 
--
Eric McCann
    University of Massachusetts, Lowell
    Department of Computer Science
    One University Avenue
    Olsen Hall, Room 304
    Lowell, MA  01854
Lab:             978.934.3385
Email:           emccann@cs.uml.edu
Homepage (lab):  www.cs.uml.edu/robots
Homepage:        www.emccann.net

**Comment by gitunit** [comment]
i have no specific subscribers so i just did a rostopic echo /tango_brain_0/depth and getting the following graph and info:

![rqt_graph](http://i.imgur.com/d8iNVkf.png)



rosnode info /rostopic_1255_1453768827119                                  
```
--------------------------------------------------------------------------------                        
Node [/rostopic_1255_1453768827119]                                                                     
Publications:                                                                                          
 * /rosout [rosgraph_msgs/Log]                                                                          
                                                                                                       
Subscriptions:                                                                                          
 * /tango_brain_0/depth [sensor_msgs/PointCloud2]                                                       
                                                                                                       
Services:                                                                                               
 * /rostopic_1255_1453768827119/set_logger_level                                                        
 * /rostopic_1255_1453768827119/get_loggers                                                             
                                                                                                        
                                                                                                        
contacting node http://xxx.xxx.xxx.xx1:60984/ ...                                                         
Pid: 1255                                                                                               
Connections:                                                                                            
 * topic: /rosout                                                                                       
    * to: /rosout                                                                                       
    * direction: outbound                                                                               
    * transport: TCPROS                                                                                 
 * topic: /tango_brain_0/depth                                                                          
    * to: /yellowstone/depth_publisher (http:///xxx.xxx.xxx.xx2:57436/)                                    
    * direction: inbound                                                                                
    * transport: TCPROS 
```
@nuclearmistake how do i check the "dead topics" ?

**Comment by nuclearmistake** [comment]
I meant the checkboxes on rqt_graph... Nothing in the past few things you've posted seems overly out if the ordinary....  How about the output of "rostopic info /tango_brain_0/depth" ?

**Comment by gitunit** [comment]
rostopic info /tango_brain_0/depth
gives me:
```                                          
Type: sensor_msgs/PointCloud2                                                                           
                                                                                                        
Publishers:                                                                                             
 * /yellowstone/depth_publisher (http://xxx.xxx.xxx.xx2:38277/)                                           
                                                                                                        
Subscribers:                                                                                            
 * /rviz_1453766993963485772 (http://xxx.xxx.xxx.xx1:32830/)                                              
 * /rostopic_1255_1453768827119 (http://xxx.xxx.xxx.xx1:60984/) 
```

**Comment by nuclearmistake** [comment]
Seems like it's completely connected... Do you see anything on /tf?

**Comment by gitunit** [comment]
nothing, it behaves the same

**Comment by gitunit** [comment]
looks like diving into the code helped me. i've found out that the variable "okPublish" was never set.
it is supposed to be set in "PeanutStream.java" in line line 145 from sharedPreferences (and as false as default in case sharedPreferences has no setting called "tb2_checked"). so i tried setting it manually to true and that did it - finally depth was publishing (at 5 Hz). so were is this setting supposed to be set? also what fixed frame do you use in rviz?

**Comment by carlosjoan91** [comment]
Thanks for the feedback, that variable used to be set via a checkbox on the
UI, it is definitely a bug (i.e., the check should disappear since the
checkbox on the UI was removed), since sharedPreferences are persistent,
mine was probably set to true from a past launch, making the bug
unreproducible for me. I will take care of removint that completely soon.
Regarding the fixed frame, that depends on your full setup, if you just
want to visualize data from the tango, I suggest turning on the "publish
map->odom" tf checkbox, then setting your fixed frame to "map" on rviz.

2016-01-30 22:02 GMT-05:00 gitunit <notifications@github.com>:

**Comment by gitunit** [comment]
@carlosjoan91 good to know. regarding the missing fixed frame, i already solved it by also setting the variable for odom to true (same bug as for the depth publisher).

how is the roadmap for raw image publishers? maybe [this](https://github.com/googlesamples/tango-examples-c/blob/master/video-overlay-jni-example/app/src/main/jni/video_overlay_app.cc) is helpful? i will try that as soon as im ready with something else...
